recipe/meta.yaml

```diff
@@ -89,25 +93,25 @@ outputs:
         # upsets MS linker and breaks pyhdk installation. On Linux
         # folly version needs version 1.78.0 so this is currently
         # windows specific.
-        - boost-cpp >1.78.0  # [win]
+        - libboost-devel  # [win]
         - llvm {{ llvm_version }}
-        - arrow-cpp ={{ arrow_version }}=*{{ build_ext }}
-        - pyarrow ={{ arrow_version }}=*{{ build_ext }}
+        # one libarrow to pick up global pinning, one to select variant
+        - libarrow
+        - libarrow =*=*{{ build_ext }}
+        - pyarrow
         - zlib
       run:
-        - arrow-cpp-proc {{ arrow_proc_version }} {{ build_ext }}
-        - pyarrow ={{ arrow_version }}=*{{ build_ext }}
+        - pyarrow =*=*{{ build_ext }}
         - python
         - openjdk 20.*
```

Comment:
Sidenote: out of curiosity, do you really need to depend on the newest java version? This is going to make hdk impossible to co-install with a relatively large set of java-related packages. Not critical, but curious (on the openjdk feedstock we mainly focus on the LTS versions, but there's been some interest lately also in the intermediate ones).

Reply:
Well, who's keeping them from depending on the latest java version on their side? That shouldn't be reflected in the recipe. Besides, java 11 & 17 still see regular releases, and will receive the same amount of security fixes.

Reply:
Let's remove this. If hdk just needs _some_ openjdk, then any consumer can add a constraint on `openjdk 20` or whatever they want.
```suggestion
        - openjdk
```
or, if there's a lower bound:

```suggestion
        - openjdk >=11
```

Reply:
cc @vlad-penkin

Reply:
I will merge this on the weekend unless there are any further comments until then.

Reply:
@h-vetinari few comments on openjdk pinning

- pyhdk targets python ecosystem, no interop with java ecosystem. openjdk is only needed to run calcite internally.
- java ecosystem use cases in conda-forge are quite limited and primarily around the bazel, which is needed for builds.
- older openjdk versions in conda-forge tends to have more vulnerabilities, even though 11 and 17 are LTS it's also about the cadence of the updates of the upstream binary builds and subsequent feedstock with repack updates
- we also want to ensure that pyhdk works with all openjdk versions in range, previously we had issues with the jni in older versions.

Reply:
> pyhdk targets python ecosystem, no interop with java ecosystem.

That's fair, though you don't know what packages users will combine in their environments; if anyone needs another package that has a dependence on an LTS openjdk, this pin would prevent them from installing that package side-by-side with hdk, which is not great.

> older openjdk versions in conda-forge tends to have more vulnerabilities, even though 11 and 17 are LTS it's also about the cadence of the updates of the upstream binary builds and subsequent feedstock with repack updates

I honestly doubt that LTS versions have more vulnerabilities, or for longer. Upstream releases builds pretty much concurrently for all supported versions every ~2 months, and the rebuilders (like temurin) that we rely on publish their versions pretty immediately afterwards. I'm also keeping an eye on this on the openjdk feedstock, and currently everything is on its newest supported version.

If you notice an issue, best would be to open an issue (or better: a PR) to the openjdk feedstock.

> we also want to ensure that pyhdk works with all openjdk versions in range, previously we had issues with the jni in older versions.

Sounds great! 👍